lib/engine/round/g_18_EU/operating.rb

```diff
@@ -0,0 +1,138 @@
+          minors, corporations = entities.partition(&:minor?)
+          corporations.select!(&:floated?)
+          corporations.sort!
+          minors + corporations
+        end
+
+        def steps
+          @current_entity.minor? ? self.class::MINOR_STEPS : self.class::STEPS
+        end
+
+        def issuable_shares
+          num_shares = @current_entity.num_player_shares - @current_entity.num_market_shares
+          bundles = @current_entity.bundles_for_corporation(@current_entity)
+          share_price = @game.stock_market.find_share_price(@current_entity, :left).price
+
+          bundles
+            .each { |bundle| bundle.share_price = share_price }
+            .reject { |bundle| bundle.num_shares > num_shares }
+        end
+
+        def redeemable_shares
+          share_price = @game.stock_market.find_share_price(@current_entity, :right).price
+
+          @game
+            .share_pool
+            .bundles_for_corporation(@current_entity)
+            .each { |bundle| bundle.share_price = share_price }
+            .reject { |bundle| @current_entity.cash < bundle.price }
+        end
+
+        private
+
+        def skip_issue
+          issuable_shares.empty? && redeemable_shares.empty?
+        end
+
+        def skip_dividend
+          return super if @current_entity.corporation?
+
+          revenue = @current_routes.sum(&:revenue)
+          process_dividend(Action::Dividend.new(
+            @current_entity,
+            kind: revenue.positive? ? 'payout' : 'withhold',
+          ))
+          true
+        end
+
+        def skip_token_or_track
+          skip_track && skip_token
+        end
+
+        def process_sell_shares(action)
+          return super if action.entity.player?
+
+          @game.share_pool.sell_shares(action.bundle)
+        end
+
+        def process_buy_shares(action)
+          @game.share_pool.buy_shares(@current_entity, action.bundle)
+        end
+
+        def payout(revenue)
+          return super if @current_entity.corporation?
+
+          @log << "#{@current_entity.name} pays out #{@game.format_currency(revenue)}"
+
+          amount = revenue / 2
+
+          [@current_entity, @current_entity.owner].each do |entity|
+            @log << "#{entity.name} receives #{@game.format_currency(amount)}"
+            @bank.spend(amount, entity)
+          end
+        end
+
+        def half(revenue)
+          withheld = revenue / 2 / 10 * 10
+          @bank.spend(withheld, @current_entity)
+          @log << "#{@current_entity.name} runs for #{@game.format_currency(revenue)} and pays half"
+          @log << "#{@current_entity.name} witholds #{@game.format_currency(withheld)}"
+          payout(revenue - withheld)
+        end
+
+        def change_share_price(_direction, revenue = 0)
+          return if @current_entity.minor?
+
+          price = @current_entity.share_price.price
+          @stock_market.move_left(@current_entity) if revenue.zero
```

Comment:
zero?

Reply:
you don't need this section, you can just do return super unless @current_entity.minor?